### ISSUE TITLE
How to train a task with soft label?

### ISSUE THREAD

**Issue by chawannut157** [nyu-mll/jiant#1337]
I'm quite new to Jiant and trying to implement knowledge distillation method.
https://arxiv.org/abs/1904.09482

Basically given soft labels (One sample): Y = [0.3,0.4,0.3] and logit(X) = [0.2,0.3,0.5]. 
The costs function can be KL divergence, cross entropy etc.

Can you suggest how to extend the framework to support such task?
First, it seems like I need to adjust this part which should not be too hard
https://github.com/nyu-mll/jiant/blob/de5437ae710c738a0481b13dc9d266dd558c43a4/jiant/proj/main/modeling/taskmodels.py#L173-L179

Then I need to change `batch.label_id.view(-1)` to support the format of soft labels.
Can you guide on this one? Is there a similar task that already do this?

**Comment by sleepinyourhat** [comment]
My impression is that the hard label format is baked in in a few places, so you'll have to do some significant investigation on your own to make this change, and it's out of scope for what we currently support. That said, @zphang might be able say more.
